src/options.c

```diff
@@ -469,6 +480,13 @@ void optionsParse(int argc, char *argv[])
                     errmsg);
             }
             break;
+        case 'w':
+            opt.windowId = optionsParseNumBase(optarg, None/*0L*/, LONG_MAX, &errmsg, 0);
```

Comment:
One concern I have is that it might be valid to have window id beyond `LONG_MAX` since `Window` is typically typedef to `unsigned long`. But I'm not sure about that.